packages/test-interface-criteria/src/criteria/TestOpportunityBookable.js

```diff
@@ -1,10 +1,18 @@
 const { complement } = require('ramda');
-const { createCriteria } = require('./criteriaUtils');
+const { getOffer, createCriteria, mustBeWithinBookingWindow } = require('./criteriaUtils');
 const { InternalTestOpportunityBookable } = require('./internal/InternalTestOpportunityBookable');
 const { supportsMinimalProposalFlow } = require('./sharedConstraints');
 
 const doesNotSupportMinimalProposalFlow = complement(supportsMinimalProposalFlow);
 
+/**
+ * @type {import('../types/Criteria').OpportunityConstraint}
+ */
+function mustBeBookable(opportunity) {
+  const offer = getOffer(opportunity);
```

Comment:
This function gets the 1st offer, but it might be the case that that offer is not applicable, but another is.

Instead, this constraint should be an OfferConstraint, rather than an OpportunityConstraint. An offer constraint is called for each offer-opportunity combo and so has (offer, opportunity) as its function signature. e.g. `supportsMinimalProposalFlow(..)`